discord/components.py

```diff
@@ -248,41 +248,48 @@ class SelectMenu(Component):
         A list of options that can be selected in this menu.
     disabled: :class:`bool`
         Whether the select is disabled or not.
+    channel_types: List[:class:`.ChannelType`]
+        A list of channel types that are allowed to be chosen in this select menu.
     """
 
     __slots__: Tuple[str, ...] = (
+        'type',
         'custom_id',
         'placeholder',
         'min_values',
         'max_values',
         'options',
         'disabled',
+        'channel_types',
     )
 
     __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
 
     def __init__(self, data: SelectMenuPayload, /) -> None:
+        self.type: ComponentType = try_enum(ComponentType, data['type'])
         self.custom_id: str = data['custom_id']
         self.placeholder: Optional[str] = data.get('placeholder')
         self.min_values: int = data.get('min_values', 1)
         self.max_values: int = data.get('max_values', 1)
         self.options: List[SelectOption] = [SelectOption.from_dict(option) for option in data.get('options', [])]
         self.disabled: bool = data.get('disabled', False)
-
-    @property
-    def type(self) -> Literal[ComponentType.select]:
-        """:class:`ComponentType`: The type of component."""
-        return ComponentType.select
+        self.channel_types: List[ChannelType] = [try_enum(ChannelType, t) for t in data.get('channel_types', [])]
 
     def to_dict(self) -> SelectMenuPayload:
         payload: SelectMenuPayload = {
             'type': self.type.value,
             'custom_id': self.custom_id,
             'min_values': self.min_values,
             'max_values': self.max_values,
-            'options': [op.to_dict() for op in self.options],
             'disabled': self.disabled,
         }
+        options = getattr(self, 'options', None)
+        if options:
+            payload['options'] = [op.to_dict() for op in options]
+
+        channel_types = getattr(self, 'channel_types', None)
+        if channel_types:
+            payload['channel_types'] = [t.value for t in channel_types]
```

Comment:
This code definitely needs work. You should take advantage of the truth-y aspect instead since right now this just looks dirty with attributes not existing that should just because of an internal detail. Using an empty list as the options or the channel_types accomplishes the same exact thing for the underlying type.